include/task.h

```diff
@@ -161,7 +161,7 @@ typedef struct xTASK_STATUS
 {
     TaskHandle_t xHandle;                         /* The handle of the task to which the rest of the information in the structure relates. */
     const char * pcTaskName;                      /* A pointer to the task's name.  This value will be invalid if the task was deleted since the structure was populated! */
-    UBaseType_t xTaskNumber;                      /* A number unique to the task. */
+    UBaseType_t xTaskNumber;                      /* A number unique to the task. Note that this is not the task number that may be modified using vTaskSetTaskNumber() and vTaskGetTaskNumber, but a separate TCB-specific and unique identifier automatically assigned on task generation. */
```

Comment:
1. uxTaskGetTaskNumber() instead of vTaskGetTaskNumber . 
2 .We should also change `xTaskNumber` to `uxTasknumber` for naming convention, since its of type `UBaseType_t` . I can update this in a separate PR.

Reply:
> uxTaskGetTaskNumber() instead of vTaskGetTaskNumber.

Good catch. Fixed.

> We should also change xTaskNumber to uxTasknumber for naming convention, since its of type UBaseType_t . I can update this in a separate PR.

My original proposal was to rename this member to `uxTCBNumber` which matches the internal implementation. The member `uxTCBNumber` in a TCB is the unique ID for each TCB, while `uxTaskNumber` is the value accessed using the above mentioned functions.

It was my understanding from  some posts buried in my discussion (here)[https://forums.freertos.org/t/tracing-improvements/20097/7] that changing the name was not an option because it broken compatibility/kernel-aware debuggers.

If a rename is on the table, I would push for making this much less confusing :)

Reply:
I saw this , thanks , yes it will be breaking change in some of the ports.